counsel.el

```diff
@@ -3254,6 +3254,35 @@ Note: Duplicate elements of `kill-ring' are always deleted."
  '(("d" counsel-yank-pop-action-remove "delete")
    ("r" counsel-yank-pop-action-rotate "rotate")))
 
+;;** `counsel-evil-registers'
+
+(defcustom counsel-evil-registers-height 5
+  "The `ivy-height' of `counsel-evil-registers'."
+  :group 'ivy
+  :type 'integer)
+
+(defun counsel-evil-registers ()
+  "Ivy replacement for `evil-show-registers'."
+  (interactive)
+  (let ((ivy-format-function #'counsel--yank-pop-format-function)
+        (ivy-height counsel-evil-registers-height))
+    (ivy-read "evil-registers: "
+              (cl-loop for (key . val) in (evil-register-list)
+                       collect
+                       (format "[%s]: %s" (char-to-string key)
+                               (or (and val (stringp val) val) "")))
```

Comment:
This can be written

```el
(format "[%c]: %s" key (if (stringp val) val ""))
```

but do we really want to include empty `val`s in the list? I don't use `evil` so I am not familiar with the semantics of `evil-register-list`.

Reply:
Thanks, that looks much better.

Yeah, we do want to include empty strings because it shows that particular register doesn't have any contents in it yet.